mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -50,6 +50,26 @@ void mlir::emitc::buildTerminatedBody(OpBuilder &builder, Location loc) {
   builder.create<emitc::YieldOp>(loc);
 }
 
+/// Check that the type of the initial value is compatible with the operations
+/// result type.
+LogicalResult verifyInitializationAttribute(Operation *op, Attribute value,
+                                            Type expectedType) {
```

Comment:
Since op is being passed in anyway for error emission, better to drop expectedType and call op->getType() inside? (would also make more meaningful original error message below accurate)